compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -538,16 +539,78 @@ private BType getFiniteTypeMatchWithIntLiteral(BLangLiteral literalExpr, BFinite
             literalExpr.value = String.valueOf(literalValue);
             return type;
         }
+        // Handle out of range ints
+        if (literalValue instanceof Double) {
+            return symTable.floatType;
+        }
+        if (literalValue instanceof String) {
+            return symTable.decimalType;
```

Comment:
Please check all the codecov warnings. Since this is a significant change, we need to pay extra attention.